File: app/src/main/java/com/originb/inkwisenote2/modules/queries/ui/QueryResultsViewModel.java
package com.originb.inkwisenote2.modules.queries.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.originb.inkwisenote2.modules.backgroundjobs.BackgroundOps;
import com.originb.inkwisenote2.modules.queries.data.QueryEntity;
import com.originb.inkwisenote2.modules.queries.data.QueryRepository;
import com.originb.inkwisenote2.modules.repositories.Repositories;
import com.originb.inkwisenote2.modules.smarthome.QueryNoteResult;
import com.originb.inkwisenote2.modules.smarthome.SmartHomePageViewModel;

import java.util.*;

public class QueryResultsViewModel extends ViewModel {
    
    private QueryRepository queryRepository;
    private SmartHomePageViewModel smartHomePageViewModel;
    
    private final MutableLiveData<List<QueryEntity>> queries = new MutableLiveData<>(new ArrayList<>());
    private final MutableLiveData<Set<QueryNoteResult>> currentQueryResults = new MutableLiveData<>(new HashSet<>());
    
    public QueryResultsViewModel() {
        queryRepository = Repositories.getInstance().getQueryRepository();
        smartHomePageViewModel = new SmartHomePageViewModel();
        
        loadQueries();
    }
    
    private void loadQueries() {
        BackgroundOps.execute(
            () -> queryRepository.getAllQueries(),
            queries::setValue
        );
    }
    
    public void loadQueryResults(String queryName) {
        BackgroundOps.execute(() -> {
            // Use the existing query results map from SmartHomePageViewModel
            Map<String, Set<QueryNoteResult>> allResults =
                smartHomePageViewModel.getLiveQueryResults().getValue();
            
            if (allResults != null && allResults.containsKey(queryName)) {
                return allResults.get(queryName);
            }
            return new HashSet<>();
        }, currentQueryResults::setValue);
    }
    
    public LiveData<List<QueryEntity>> getQueries() {
        return queries;
    }
    
    public LiveData<Set<QueryNoteResult>> getCurrentQueryResults() {
        return currentQueryResults;
    }
} 

File: androidApp/src/main/java/com/originb/inkwisenote2/modules/queries/ui/QueryResultsViewModel.kt
package com.originb.inkwisenote2.modules.queries.ui

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.originb.inkwisenote2.modules.backgroundjobs.BackgroundOps
import com.originb.inkwisenote2.modules.queries.data.QueryEntity
import com.originb.inkwisenote2.modules.queries.data.QueryRepository
import com.originb.inkwisenote2.modules.smarthome.QueryNoteResult
import com.originb.inkwisenote2.modules.smarthome.SmartHomePageViewModel
import java.util.concurrent.Callable
import java.util.function.Consumer

class QueryResultsViewModel(
    private val queryRepository: QueryRepository,
    private val smartHomePageViewModel: SmartHomePageViewModel
) : ViewModel() {
    private val queries = MutableLiveData<MutableList<QueryEntity>>(ArrayList())
    private val currentQueryResults = MutableLiveData<MutableSet<QueryNoteResult>>(HashSet())

    init {
        loadQueries()
    }

    private fun loadQueries() {
        BackgroundOps.execute(
            { queryRepository.allQueries },
            Consumer { queries.value = it }
        )
    }

    fun loadQueryResults(queryName: String) {
        BackgroundOps.execute(
            {
                val allResults = smartHomePageViewModel.getLiveQueryResults().value!!
                allResults[queryName]?.toMutableSet()
            },
            { results ->
                currentQueryResults.value = results ?: mutableSetOf()
            }
        )
    }

    fun getQueries(): LiveData<MutableList<QueryEntity>> {
        return queries
    }

    fun getCurrentQueryResults(): LiveData<MutableSet<QueryNoteResult>> {
        return currentQueryResults
    }
}